cli.ts

```diff
@@ -1,14 +1,14 @@
 import type {UnvalidatedPluginInfo, InstalledPlugin, Config, ConfigArgs, Alias, Verb, Option, PositionalArg, PluginResponse, PluginAction, ProxyAction} from './taqueria-protocol/taqueria-protocol-types.ts'
 import {Task, PluginInfo} from './taqueria-protocol/taqueria-protocol-types.ts'
-import type {EnvKey, EnvVars, DenoArgs, RawInitArgs, SanitizedInitArgs, i18n, InstallPluginArgs, UninstallPluginArgs} from './taqueria-types.ts'
+import type {EnvKey, EnvVars, DenoArgs, RawInitArgs, SanitizedInitArgs, i18n, InstallPluginArgs, UninstallPluginArgs, SanitizedScaffoldInitArgs} from './taqueria-types.ts'
```

Comment:
The name of `RawInitArgs` and `SanitizedInitArgs` is misleading. I should have named them 'RawArgs' and 'SanitizedArgs', as they aren't specific to the `init` task. To be consistent with the other named types, I should have probably named these "UnvalidedTaskArgs" and "TaskArgs" - that would follow the naming convention used elsewhere.

In any case, what's your thoughts on just consolidating SanitizedInitArgs and SanitizedScaffoldInitArgs?

Although having separate types for each built-in task might seem appealing, it might just add noise - yargs parsing of ALL args is done atomically, so my thinking here is that we just need a type to represent the args after they were parsed by yargs, and then another type to represent those args after we've done validation and sanitization for our limited set of built-in tasks. 

I could probably go either way on this though, so I'd be curious what your opinion is. If we do think having separate types for sanitized args specific to each built-in task makes sense, then I should probably refactor and segment SanitizedInitArgs, as right now the intent was that this type would be a representation of sanitized args after yargs parsed them, and that these args are valid and can be used by any built-in task.

Reply:
The reason I had to separate those is because of the difference between optional types:

```
export interface SanitizedInitArgs {
    scaffoldUrl?: SanitizedUrl;
    scaffoldProjectDir?: SanitizedAbsPath;
}
export interface SanitizedScaffoldInitArgs extends SanitizedInitArgs {
    scaffoldUrl: SanitizedUrl;
    scaffoldProjectDir: SanitizedAbsPath;
}
const sanitizeScaffoldArgs = (parsedArgs: RawInitArgs) : SanitizedScaffoldInitArgs => ({
    ...sanitizeArgs(parsedArgs),
     scaffoldUrl: SanitizedUrl.create(parsedArgs.scaffoldUrl ?? ''),
     scaffoldProjectDir: SanitizedAbsPath.create(parsedArgs.scaffoldProjectDir ?? ''),
})
```

Only the `sanitizeScaffoldArgs` can guarantee that those values actually exist.

Any thoughts on a better pattern here?